rust/stake_neuron_from_cli/README.md

```diff
@@ -0,0 +1,179 @@
+- Domain length plus domain id: `b"\x0Aaccount-id"`
+
+### 3. **Two-Step Neuron Creation Process**
+
+See the complete neuron staking flow:
+
+1. **Transfer ICP** → Transfer tokens to governance canister's computed subaccount
+2. **Claim Neuron** → Call `claim_or_refresh_neuron_from_account` to create the neuron
+
+```rust
+// Step 1: Transfer ICP to the computed subaccount
+let to_account = AccountIdentifier::new(&governance_principal, Some(subaccount));
+let transfer_result = ledger.transfer(TransferArgs {
+    memo: nonce,
+    amount: Tokens { e8s: amount },
+    to: to_account.to_vec(),
+    // ...
+}).await?;
+
+// Step 2: Claim the neuron using the same nonce
+let claim_result = governance.claim_or_refresh_neuron_from_account(
+    ClaimOrRefreshNeuronFromAccount {
+        controller: Some(controller),
+        memo: nonce,
+    }
+).await?;
+```
+
+### 4. **Identity Management with ic-agent**
+
+Additionally, you can see how you can use ic-agent to use different key formats to send messages.
+
+```rust
+async fn load_identity(identity_path: &PathBuf) -> Result<Box<dyn Identity>, Box<dyn std::error::Error>> {
+    let pem_content = std::fs::read_to_string(identity_path)?;
+    
+    // Try different identity formats
+    if let Ok(identity) = BasicIdentity::from_pem(pem_content.as_bytes()) {
+        return Ok(Box::new(identity));
+    }
+    
+    if let Ok(identity) = Secp256k1Identity::from_pem(pem_content.as_bytes()) {
+        return Ok(Box::new(identity));
+    }
+    
+    Err("Could not parse identity file".into())
+}
+```
+
+## Local Testing with setup_and_run.sh
+
+The `setup_and_run.sh` script provides a complete local testing environment:
+
+```bash
+#!/bin/bash
+# 1. Start local IC replica
+dfx start --clean --background
+
+# 2. Deploy NNS canisters with canonical IDs
+dfx deploy icp_ledger --specified-id ryjl3-tyaaa-aaaaa-aaaba-cai
+dfx deploy nns_governance --specified-id rrkah-fqaaa-aaaaa-aaaaq-cai
+
+# 3. Run the CLI tool
+cargo run -- --identity "$HOME/.config/dfx/identity/default/identity.pem" \
+              --url "http://127.0.0.1:4943" \
+              --amount 100010000 \
+              --nonce 42
+```
+
+**What this teaches:**
+- How to set up local NNS environment
+- Importance of canonical canister IDs for proper inter-canister calls
+- Complete governance and ledger initialization arguments
+- Integration between `dfx` and Rust tooling
+
+
+## Running the Example
+
+### Prerequisites
+
+1. **Install dfx**: Follow [DFINITY SDK installation](https://internetcomputer.org/docs/current/developer-docs/setup/install/)
+2. **Rust toolchain**: `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs/ | sh`
+
+### Option 1: Manual CLI Usage
+
+```bash
+# Build the CLI
+cargo build --release
+
+# Run with your own identity 
+./target/release/stake_neuron_from_cli \
+    --identity ~/.config/dfx/identity/default/identity.pem \
+    --url [YOUR LOCAL TEST ENVIRONMENT URL] \
+    --amount 100010000 \
+    --nonce 42
+```
```

Comment:
i'm not sure this option is required. when would i use it? if i don't deploy the canisters first, it won't work. also, if i don't use the identity that was minted the the initial icp balance, it won't work.